pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1419,40 +1428,38 @@ private ByteBuf processMessageChunk(ByteBuf compressedPayload, MessageMetadata m
      * Notify waiting asyncReceive request with the received message.
      *
      * @param message
+     * @return the message is enqueue incomingMessages.
      */
-    void notifyPendingReceivedCallback(final Message<T> message, Exception exception) {
-        if (pendingReceives.isEmpty()) {
```

Comment:
here remove it to avoid thread race.

Reply:
Can you clarify which race you're referencing? If you're referring to the implementation of `hasNextPendingReceive()`, `pendingReceives` is only supposed to be updated on the `internalPinnedExecutor`, so the check is redundant, but is not necessarily a race. (We can still remove the check, but I just want to see if there is another race you noticed.)

Reply:
Hi, I just review the code and felt that the judgment in this place might be unnecessary, introducing the risk of thread races. I double checked, it's single scheduled executor, not cause thread race.